src/emitters/tests/test_distant.py

```diff
@@ -0,0 +1,119 @@
+import mitsuba
+import pytest
+import enoki as ek
+import numpy as np
+
+
+spectrum_strings = {
+    'd65': """
+        <spectrum version='2.0.0' name="irradiance" type='d65'/>
+        """,
+    'regular': """
+        <spectrum version='2.0.0' name="irradiance" type='regular'>
+           <float name="lambda_min" value="500"/>
+           <float name="lambda_max" value="600"/>
+           <string name="values" value="1, 2"/>
+        </spectrum>
+        """,
+    'null': """
+        <spectrum version='2.0.0' type='uniform'>
+            <float name="value" value="0"/>
+        </spectrum>
+        """,
+}
+
+spectrum_keys = list(set(spectrum_strings.keys()) - {"null"})
+
+
+def make_spectrum(spectrum_key="d65"):
+    from mitsuba.core.xml import load_string
+
+    spectrum = load_string(spectrum_strings[spectrum_key])
+    expanded = spectrum.expand()
+
+    if len(expanded) == 1:
+        spectrum = expanded[0]
+
+    return spectrum
+
+
+def make_emitter(direction="0,0,-1", spectrum_key="d65"):
+    from mitsuba.core.xml import load_string
+
+    if type(direction) is not str:
+        direction = " ".join([str(x) for x in direction])
+
+    return load_string("""
+        <emitter version="2.0.0" type="distant">
+            <vector name="direction" value="{d}"/>
+            {s}
+        </emitter>
+    """.format(d=direction, s=spectrum_strings[spectrum_key]))
+
+
+def test_construct(variant_scalar_rgb):
+    # Test if the emitter can be constructed as intended
+    e = make_emitter()
+    assert not e.bbox().valid()  # Degenerate bounding box
+    assert ek.allclose(
+        e.world_transform().eval(0.).matrix,
+        [[0, 1, 0, 0],
+         [1, 0, 0, 0],
+         [0, 0, -1, 0],
+         [0, 0, 0, 1]]
+    )
+
+
+@pytest.mark.parametrize("spectrum_key", spectrum_keys)
+def test_eval(variant_scalar_spectral, spectrum_key):
+    # Check correctness of the eval() method
+    from mitsuba.render import SurfaceInteraction3f
+
+    direction = np.array([0, 0, -1])
+    emitter = make_emitter(direction, spectrum_key)
+    spectrum = make_spectrum(spectrum_key)
+
+    # Incident direction in the illuminated direction
+    wi = [0, 0, 1]
+    it = SurfaceInteraction3f()
+    it.p = [0, 0, 0]
+    it.wi = wi
+    assert ek.allclose(emitter.eval(it), spectrum.eval(it))
+
+    # Incident direction off the illuminated direction
+    null_spectrum = make_spectrum("null")
+    wi = [0, 0, 1.1]
+    it = SurfaceInteraction3f()
+    it.p = [0, 0, 0]
+    it.wi = wi
+    assert ek.allclose(emitter.eval(it), null_spectrum.eval(it))
+
+
+@pytest.mark.parametrize("spectrum_key", spectrum_keys)
+def test_sample_direction(variant_scalar_spectral, spectrum_key):
+    # Check correctness of sample_direction() and pdf_direction() methods
+
+    from mitsuba.render import SurfaceInteraction3f
+
+    direction = np.array([0, 0, -1])
```

Comment:
Maybe do a `for` loop over a couple directions here?

Reply:
I parametrised the tests with a set of arbitrary directions.